src/knot_cloud.c

```diff
@@ -638,16 +638,21 @@ int knot_cloud_list_devices(void)
  *
  * Returns: 0 if successful and a KNoT error otherwise.
  */
-int knot_cloud_publish_data(const char *id, uint8_t sensor_id,
-			    uint8_t value_type, const knot_value_type *value,
-			    uint8_t kval_len)
+int knot_cloud_publish_data(const char *id, int sensor_id,
+			    int value_type, void *value,
+			    int kval_len)
 {
 	json_object *jobj_data;
 	const char *json_str;
 	int result;
 
-	jobj_data = parser_data_create_object(id, sensor_id, value_type, value,
-					      kval_len);
+	knot_value_type *t_value = (knot_value_type *) value;
+	uint8_t t_sensor_id = (uint8_t) sensor_id;
```

Comment:
i think this cast is dangerous because it could cause a overflow. the 'int' type can vary depending on cpu architecture from 8 to 32 bits.

Reply:
> i think this cast is dangerous because it could cause a overflow. the 'int' type can vary depending on cpu architecture from 8 to 32 bits.

Also, i think the way it was before is a good representation on how this API works.

Reply:
You can use unsigned char since the macro uint8_t expands for unsigned char.

Reply:
this refers to an outdated version of the code, right? currently he's using an "unsigned char" for the argument, which makes this a safe cast.

also, I'd like to better understand why using "unsigned char" would presents any advantage over "uint8_t", since knot code as a whole uses it instead of "unsigned char".

Reply:
The problem with "uint8_t" is that it creates a dependency on the knot_cloud header, but with go to the library where "uint8_t" is declared, its a "unsigned char".